commands/command_migrate_info.go

```diff
@@ -0,0 +1,161 @@
+package commands
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/git-lfs/git-lfs/errors"
+	"github.com/git-lfs/git-lfs/git/odb"
+	"github.com/git-lfs/git-lfs/tools"
+	"github.com/git-lfs/git-lfs/tools/humanize"
+	"github.com/spf13/cobra"
+)
+
+var (
+	// migrateInfoTopN is a flag given to the git-lfs-migrate(1) subcommand
+	// 'info' which specifies how many info entries to show by default.
+	migrateInfoTopN int
+
+	// migrateInfoAboveFmt is a flag given to the git-lfs-migrate(1)
+	// subcommand 'info' specifying a human-readable string threshold of
+	// filesize before entries are counted.
+	migrateInfoAboveFmt string
+	// migrateInfoAbove is the number of bytes parsed from the above
+	// migrateInfoAboveFmt flag.
+	migrateInfoAbove uint64
+)
+
+func migrateInfoCommand(cmd *cobra.Command, args []string) {
+	exts := make(map[string]*MigrateInfoEntry)
+
+	above, err := humanize.ParseBytes(migrateInfoAboveFmt)
+	if err != nil {
+		ExitWithError(errors.Wrap(err, "cannot parse --above=<n>"))
+	}
+
+	migrateInfoAbove = above
+
+	migrate(cmd, args, func(path string, b *odb.Blob) (*odb.Blob, error) {
+		ext := fmt.Sprintf("*%s", filepath.Ext(path))
+
+		if len(ext) > 1 {
+			entry := exts[ext]
+			if entry == nil {
+				entry = &MigrateInfoEntry{Qualifier: ext}
+			}
+
+			entry.Total++
+			entry.BytesTotal += b.Size
+
+			if b.Size > int64(migrateInfoAbove) {
+				entry.TotalAbove++
+				entry.BytesAbove += b.Size
+			}
+
+			// TODO(@ttaylorr): needed?
+			exts[ext] = entry
+		}
+
+		return b, nil
+	})
+
+	entries := EntriesBySize(MapToEntries(exts))
+	sort.Sort(sort.Reverse(entries))
+
+	migrateInfoTopN = tools.ClampInt(migrateInfoTopN, len(entries), 0)
+
+	entries = entries[:tools.MaxInt(0, migrateInfoTopN)]
+
+	entries.Print(os.Stderr)
+}
+
+// MigrateInfoEntry represents a tuple of filetype to total size taken by that
+// file type.
```

Comment:
Nice catch! Updated those docs and added some godoc to the fields on this type in 7c77064.